File: app/src/main/java/com/example/techfix/data/dao/DeviceCategoryDao.java
package com.example.techfix.data.dao;
import androidx.room.*;
import com.example.techfix.data.entity.DeviceCategory;
import java.util.List;
@Dao public interface DeviceCategoryDao {
 @Insert(onConflict=OnConflictStrategy.REPLACE) void insertCategory(DeviceCategory category);
 @Query("SELECT * FROM device_categories") List<DeviceCategory> getAllCategories();
}

File: app/src/main/java/com/example/techfix/data/dao/DeviceCategoryDao.kt
package com.example.techfix.data.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.techfix.data.entity.DeviceCategory

@Dao
interface DeviceCategoryDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertCategory(category: DeviceCategory)

    @Query("SELECT * FROM device_categories")
    suspend fun getAllCategories(): List<DeviceCategory>
}